source/lac/petsc_parallel_block_sparse_matrix.cc

```diff
@@ -31,6 +32,11 @@ namespace PETScWrappers
       return *this;
     }
 
+    BlockSparseMatrix::~BlockSparseMatrix()
+    {
+      PetscErrorCode ierr = destroy_matrix(petsc_nest_matrix);
+      AssertThrow(ierr == 0, ExcPETScError(ierr));
```

Comment:
```suggestion
      AssertNothrow(ierr == 0, ExcPETScError(ierr));
```
No throwing from dtors

FYI: like the other PR: we do have `destroy_matrix()` but it doesn't do anything but call `MatDestroy()` these days. It was written at a time when we supported old versions of PETSc which had a different API. I'd prefer to just remove the function and call `MatDestroy()` directly.